src/main/java/pl/edu/mimuw/chess/AbstractPiece.java

```diff
@@ -0,0 +1,49 @@
+package pl.edu.mimuw.chess;
+
+import java.util.List;
+
+import static pl.edu.mimuw.chess.V2.*;
+
+public abstract class AbstractPiece implements ChessPiece {
+	private V2 position;
+	private final ChessColor color;
+	private final String whiteRepresentation;
+	private final String blackRepresentation;
+	private final List<List<V2>> POSSIBLE_MOVES;
+	
+	protected AbstractPiece(V2 position, ChessColor color, String whiteRepresentation,
+	                        String blackRepresentation) {
+		this.position = position;
+		this.color = color;
+		this.whiteRepresentation = whiteRepresentation;
+		this.blackRepresentation = blackRepresentation;
+		this.POSSIBLE_MOVES = this.generatePossibleMoves();
```

Comment:
a to jest antypattern i zdecydowanie tak nie należy robić - problem jest w tym, że wołasz metodę klasową w konstruktorze, gdzie obiekt moglby nie byc jeszcz zainicjowany, a metoda moglaby sie do czegs moze odwolac (u Ciebie tak nie ma, ale chodzi o podejscie). Imo lepiej dac te mozliwe ruchy jako parametr konstruktora, a wtedy podkalsy mobglyby to inicjowac ze statycznej metody lub nawet statycznego pola klasowego (które byloby liczone tylko raz)